lib/blemulator/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -123,26 +136,76 @@ class PlatformToDartBridge {
           arguments[SimulationArgumentName.serviceId],
           arguments[SimulationArgumentName.characteristicUuid],
         )
+        .then((characteristicResponse) => _convertToMap(
+              arguments[SimulationArgumentName.deviceIdentifier],
+              characteristicResponse.characteristic,
+              characteristicResponse.value,
+            ));
+  }
+
+  Future<dynamic> _writeCharacteristicForIdentifier(MethodCall call) async {
+    Map<dynamic, dynamic> arguments = call.arguments;
+    return _manager
+        ._writeCharacteristicForIdentifier(
+          call.arguments[SimulationArgumentName.characteristicIdentifier],
+          call.arguments[SimulationArgumentName.value],
+        )
+        .then((characteristicResponse) => _convertToMap(
+              arguments[SimulationArgumentName.deviceIdentifier],
+              characteristicResponse,
+              arguments[SimulationArgumentName.value],
+            ));
+  }
+
+  Future<dynamic> _writeCharacteristicForDevice(MethodCall call) async {
+    Map<dynamic, dynamic> arguments = call.arguments;
+    return _manager
+        ._writeCharacteristicForDevice(
+          arguments[SimulationArgumentName.deviceIdentifier],
+          arguments[SimulationArgumentName.serviceUuid],
+          arguments[SimulationArgumentName.characteristicUuid],
+          arguments[SimulationArgumentName.value],
+        )
+        .then((characteristic) => _convertToMap(
+              arguments[SimulationArgumentName.deviceIdentifier],
+              characteristic,
+              arguments[SimulationArgumentName.value],
+            ));
+  }
+
+  Future<dynamic> _writeCharacteristicForService(MethodCall call) async {
+    Map<dynamic, dynamic> arguments = call.arguments;
+    return _manager
+        ._writeCharacteristicForService(
+          arguments[SimulationArgumentName.serviceId],
+          arguments[SimulationArgumentName.characteristicUuid],
+          arguments[SimulationArgumentName.value],
+        )
         .then((characteristic) => _convertToMap(
-            arguments[SimulationArgumentName.deviceIdentifier],
-            characteristic));
+              arguments[SimulationArgumentName.deviceIdentifier],
+              characteristic,
+              arguments[SimulationArgumentName.value],
+            ));
   }
 
   Map<String, dynamic> _convertToMap(
-          String peripheralId, CharacteristicResponse response) =>
+    String peripheralId,
```

Comment:
Why is peripheralId always sent to platform, even in discovery?

Reply:
For the sake of consistency. We've already discussed it, so Im resolving this conversation.